modules/internal/ChapelRange.chpl

```diff
@@ -514,7 +514,58 @@ module ChapelRange {
   proc ident(r1: range(?), r2: range(?)) param
     return false;
   
-  
+  //////////////////////////////////////////////////////////////////////////////////
+  // Range Casts
+  //
+/* Cast a range to another range type. If the old type is stridable and the
+   new type is not stridable, ensure that the old stride was 1.
+ */
+pragma "no doc"
+proc range.safeCast(type t) where isRangeType(t) {
+  var tmp: t;
+  if tmp.stridable {
+    tmp._low = this.low;
+    tmp._high = this.high;
+    tmp._stride = this.stride;
+    tmp._alignment = this.alignment;
+    tmp._aligned = this.aligned;
+    return tmp;
+  } else {
+    if this.stride == 1 {
+      tmp._low = this.low;
+      tmp._high = this.high;
+      tmp._alignment = this.alignment;
+      tmp._aligned = this.aligned;
+      return tmp;
+    } else {
+      halt("illegal safeCast from non-unit stride range to unstridable range");
+    }
+  }
+}
+
+/* Cast a range to a new range type.  If the old type was stridable and the
+   new type is not stridable, then assume the stride was 1 without checking.
```

Comment:
In the above comment, how about instead of "assume stride 1 without checking" we say "force the stride to be 1".